src/main/java/com/aws/greengrass/disk/spool/DiskSpoolDAO.java

```diff
@@ -235,51 +284,96 @@ protected void setUpDatabase() throws SQLException {
                 + "contentType STRING"
                 + ");";
         DriverManager.registerDriver(new org.sqlite.JDBC());
-        try (Connection conn = getDbInstance();
-            Statement st = conn.createStatement()) {
-            //create new table if table doesn't exist
-            st.executeUpdate(tableCreationString);
-        }
+        performSqlOperation(conn -> {
+            try (Statement st = conn.createStatement()) {
+                // create new table if table doesn't exist
+                st.executeUpdate(query);
+                return null;
+            }
+        }, "create-spooler-table");
     }
 
-    private SpoolMessage getSpoolMessageFromRs(long messageId, ResultSet rs) throws SQLException, IOException {
-        if (rs.next()) {
-            Publish request = Publish.builder()
-                    .qos(QOS.fromInt(rs.getInt("qos")))
-                    .retain(rs.getBoolean("retain"))
-                    .topic(rs.getString("topic"))
-                    .payload(rs.getBytes("payload"))
-                    .payloadFormat(rs.getObject("payloadFormat") == null
-                            ? null : Publish.PayloadFormatIndicator.fromInt(rs.getInt("payloadFormat")))
-                    .messageExpiryIntervalSeconds(rs.getObject("messageExpiryIntervalSeconds") == null
-                            ? null : rs.getLong("messageExpiryIntervalSeconds"))
-                    .responseTopic(rs.getString("responseTopic"))
-                    .correlationData(rs.getBytes("correlationData"))
-                    .contentType(rs.getString("contentType"))
-                    .userProperties(rs.getString("userProperties") == null
-                            ? null : mapper.readValue(rs.getString("userProperties"),
-                                    new TypeReference<List<UserProperty>>(){})).build();
-
-            return SpoolMessage.builder()
-                    .id(messageId)
-                    .retried(new AtomicInteger(rs.getInt("retried")))
-                    .request(request).build();
-        } else {
-            return null;
+    @SuppressWarnings("PMD.AvoidCatchingGenericException")
+    private <T> T performSqlOperation(CrashableFunction<Connection, T, SQLException> operation,
+                                      String operationName) throws InterruptedException, SQLException {
+        try {
+            return RetryUtils.runWithRetry(
```

Comment:
why do we run with a retry? This isn't retrying in the case that the DB is corrupted, so what exactly are we retrying?

Reply:
yeah, `SQLTransientConnectionException` is pretty vague--looking at its subclasses, it can happen for timeouts (unlikely unless the db is super overloaded and slow device?) and transaction rollback errors (n/a since we dont use transactions).  ofc this might not be every situation

Reply:
to your point though, spooler interface can throw an exception, so callers should be prepared to retry. either way, can remove retries in a follow up PR since it's existing behavior

Reply:
jk will do it here since it cleans up all the InterruptedException stuff